shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +96,419 @@ public static void addMetaTag(
         messageHeader.setMeta(meta);
     }
 
-    public static void setMessageTypeCoding(Bundle order, Coding coding) {
-        var messageHeader = findOrInitializeMessageHeader(order);
+    // MSH-3 - Sending Application
+    public static MessageHeader.MessageSourceComponent getMSH3MessageSourceComponent(
+            Bundle bundle) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        return messageHeader.getSource();
+    }
+
+    public static void setMSH3MessageSourceComponent(
+            Bundle bundle, MessageHeader.MessageSourceComponent sendingApplication) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        messageHeader.setSource(sendingApplication);
+    }
+
+    public static MessageHeader.MessageSourceComponent createMSH3MessageSourceComponent() {
+        MessageHeader.MessageSourceComponent source = new MessageHeader.MessageSourceComponent();
+        source.setId(UUID.randomUUID().toString());
+        return source;
+    }
+
+    // MSH-4 - Sending Facility
+    public static Organization getMSH4Organization(Bundle bundle) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        return (Organization) messageHeader.getSender().getResource();
+    }
+
+    public static void setMSH4Organization(Bundle bundle, Organization sendingFacility) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        Reference organizationReference = createOrganizationReference(bundle, sendingFacility);
+        messageHeader.setSender(organizationReference);
+    }
+
+    // MSH-4.1 - Namespace ID
+    public static Identifier getMSH4_1Identifier(Bundle bundle) {
+        Organization sendingFacility = getMSH4Organization(bundle);
+        if (sendingFacility == null) {
+            return null;
+        }
+        List<Identifier> identifiers = sendingFacility.getIdentifier();
+        return getHD1Identifier(identifiers);
+    }
+
+    // MSH-5 - Receiving Application
+    public static MessageHeader.MessageDestinationComponent getMSH5MessageDestinationComponent(
+            Bundle bundle) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        return messageHeader.getDestinationFirstRep();
+    }
+
+    public static void setMSH5MessageDestinationComponent(
+            Bundle bundle, MessageHeader.MessageDestinationComponent receivingApplication) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        messageHeader.setDestination(List.of(receivingApplication));
+    }
+
+    // MSH-6 - Receiving Facility
+    public static Organization getMSH6Organization(Bundle bundle) {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.

Reply:
Moved to test package